src/fsharp/FSharp.Core/fslib-extra-pervasives.fs

```diff
@@ -78,6 +78,13 @@ module ExtraTopLevelOperators =
                 if t.ContainsKey(safeKey) then (r <- t.[safeKey]; true) else false
             member s.Remove(_ : 'Key) = (raise (NotSupportedException(SR.GetString(SR.thisValueCannotBeMutated))) : bool) 
 
+        interface IReadOnlyDictionary<'Key, 'T> with
+            member __.Item with get key = t.[makeSafeKey key]
+            member __.Keys = t.Keys |> Seq.map getKey
+            member __.TryGetValue(key, value) = t.TryGetValue(makeSafeKey key, ref value)
```

Comment:
``ref value`` is incorrect here - please use ``&value``.  Please also add a test for this and review all testing as there must be a test hole here.

Reply:
Ref seems to work fine for me? I've added tests to validate it - I'll use the `&` instead though.

Reply:
It is my understanding that ``ref`` will not copy-out the resulting value to the byref passed as an argument